saleor/order/calculations.py

```diff
@@ -345,22 +345,21 @@ def _recalculate_with_plugins(
 
 
 def _get_undiscounted_price(
-    line_price: OrderTaxedPricesData,
-    line_base_price: Money,
+    price: OrderTaxedPricesData,
+    undiscounted_base_price: Money,
     tax_rate,
     prices_entered_with_tax,
-):
-    if (
-        tax_rate > 0
-        and line_price.undiscounted_price.net == line_price.undiscounted_price.gross
-    ):
-        get_taxed_undiscounted_price(
-            line_base_price,
-            line_price.undiscounted_price,
-            tax_rate,
-            prices_entered_with_tax,
+) -> TaxedMoney:
+    if tax_rate > 0 and price.undiscounted_price.net == price.undiscounted_price.gross:
+        return quantize_price(
+            calculate_flat_rate_tax(
```

Comment:
A few things are unclear to me:
- before, `get_taxed_undiscounted_price` was missing the return statement which was obviously wrong
- now we're replacing `get_taxed_undiscounted_price` with `calculate_flat_rate_tax` which was done anyway in `get_taxed_undiscounted_price`, why are we doing this? Is there something wrong with the implementation of `get_taxed_undiscounted_price` which looks like should be used here?

Reply:
Just adding the missing return statement will not solve the issue because inside `get_taxed_undiscounted_price` function, this comparison: `if undiscounted_base_price == price_to_compare:` is always `true`.
Since the `get_taxed_undiscounted_price` function is used in checkout flow (where everything works correctly) I didn't want to refactor the function.
Does it make sense?

Reply:
Is it a different use case in this place vs. other usages in checkout where this comparison makes sense? I didn't check but I'm wondering if we're sure about the use-cases.

Reply:
OK, now I see we can reuse the `get_taxed_undiscounted_price` function, but we need to pass  `price=line_price.price_with_discounts` instead of `price=line_price.undiscounted_price`. Let me rewrite it